src/libYARP_OS/src/BottleImpl.cpp

```diff
@@ -817,21 +817,6 @@ ConstString StoreDouble::toString() const
     } else {
         str += ".0";
     }
-
-    int ct = 0;
-    for (size_t i = str.length(); i >= 1; i--) {
-        if (str[i - 1] != '0') {
-            if (str[i - 1] == '.') {
-                ct--;
-                i++;
-            }
-            if (ct >= 1) {
-                str = str.substr(0, i);
-            }
-            break;
-        }
-        ct++;
```

Comment:
Why this disappeared?

Reply:
By my understanding (cc @Nicogene) `sprintf(buf, "%.*g", DBL_DIG, x);` with the specific `%.*g` argument automatically accounts for removing trailing zeros.

Reply:
right! That code was manually removing trailing zeros, while `%g` does not print them in the first place.

Reply:
Ok, thanks!